wcfsetup/install/files/ts/WoltLabSuite/Core/Ui/Dropdown/Simple.ts

```diff
@@ -0,0 +1,596 @@
+      _activeTargetId = targetId;
+      dropdown.classList.add('dropdownOpen');
+      menu.classList.add('dropdownOpen');
+
+      const button = dropdown.querySelector('.dropdownToggle');
+      if (button) button.setAttribute('aria-expanded', 'true');
+
+      const list: HTMLElement | null = menu.childElementCount > 0 ? menu.children[0] as HTMLElement : null;
+      if (list && Core.stringToBool(list.dataset.scrollToActive || '')) {
+        delete list.dataset.scrollToActive;
+
+        let active: HTMLElement | null = null;
+        for (let i = 0, length = list.childElementCount; i < length; i++) {
+          if (list.children[i].classList.contains('active')) {
+            active = list.children[i] as HTMLElement;
+            break;
+          }
+        }
+
+        if (active) {
+          list.scrollTop = Math.max((active.offsetTop + active.clientHeight) - menu.clientHeight, 0);
+        }
+      }
+
+      const itemList = menu.querySelector('.scrollableDropdownMenu');
+      if (itemList !== null) {
+        itemList.classList[(itemList.scrollHeight > itemList.clientHeight ? 'add' : 'remove')]('forceScrollbar');
+      }
+
+      notifyCallbacks(containerId, 'open');
+
+      if (!disableAutoFocus) {
+        menu.setAttribute('role', 'menu');
+        menu.tabIndex = -1;
+        menu.removeEventListener('keydown', dropdownMenuKeyDown);
+        menu.addEventListener('keydown', dropdownMenuKeyDown);
+        menu.querySelectorAll('li').forEach(listItem => {
+          if (!listItem.clientHeight) return;
+          if (firstListItem === null) firstListItem = listItem;
+          else if (listItem.classList.contains('active')) firstListItem = listItem;
+
+          listItem.setAttribute('role', 'menuitem');
+          listItem.tabIndex = -1;
+        });
+      }
+
+      UiDropdownSimple.setAlignment(dropdown, menu, alternateElement);
+
+      if (firstListItem !== null) {
+        firstListItem.focus();
+      }
+    }
+  });
+
+  window.WCF.Dropdown.Interactive.Handler.closeAll();
+
+  return (event === null);
+}
+
+function handleKeyDown(event: KeyboardEvent): void {
+  // <input> elements are not valid targets for drop-down menus. However, some developers
+  // might still decide to combine them, in which case we try not to break things even more.
+  const target = event.currentTarget as HTMLElement;
+  if (target.nodeName === 'INPUT') {
+    return;
+  }
+
+  if (event.key === 'Enter' || event.key === 'Space') {
+    event.preventDefault();
+    toggle(event);
+  }
+}
+
+function dropdownMenuKeyDown(event: KeyboardEvent): void {
+  let button, dropdown;
+
+  const activeItem = document.activeElement as HTMLElement;
+  if (activeItem.nodeName !== 'LI') {
+    return;
+  }
+
+  if (event.key === 'ArrowDown' || event.key === 'ArrowUp' || event.key === 'End' || event.key === 'Home') {
+    event.preventDefault();
+
+    const listItems = Array.from<HTMLElement>(activeItem.closest('.dropdownMenu')!.querySelectorAll('li'));
```

Comment:
```suggestion
    const listItems: HTMLElement[] = Array.from(activeItem.closest('.dropdownMenu')!.querySelectorAll('li'));
```
Reason: Abusing generics to get a proper return type is discouraged.